segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollector.java

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.chunklayer;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.AbstractThreadPoolService;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.concurrent.Services;
+import io.pravega.segmentstore.storage.metadata.ChunkMetadata;
+import lombok.Data;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Implements simple garbage collector for cleaning up the deleted chunks.
+ * The garbage collector maintains a in memory queue of chunks to delete which is drained by a background task.
+ * This queue is populated by following
+ * 1. Various SLTS operations requesting deletes
+ * 2. Background task that scans all records to find undeleted chunks inside metadata (not yet implemented).
+ * 3. Background task that scans all LTS to find unaccounted chunks that are deemed garbage (not yet implemented).
```

Comment:
Please, create 2 Git issues to keep track of these problems in the future and add them to the comments.
Also, the "scans all records" part sounds scary in terms of efficiency, we need to think of a better approach to do that.